File: src/snippets/rate-limiting/reference/bun/ByUserId.ts
import arcjet, { fixedWindow } from "@arcjet/bun";
import { env } from "bun";

const aj = arcjet({
  key: env.ARCJET_KEY!, // Get your site key from https://app.arcjet.com
  rules: [
    fixedWindow({
      mode: "LIVE",
      characteristics: ["userId"],
      window: "1h",
      max: 60,
    }),
  ],
});

export default {
  port: 3000,
  fetch: aj.handler(async (req) => {
    // Pass userId as a string to identify the user. This could also be a number
    // or boolean value
    const decision = await aj.protect(req, { userId: "user123" });
    console.log("Arcjet decision", decision);

    if (decision.isDenied()) {
      return new Response("Too many requests", { status: 429 });
    }
    return new Response("Hello world");
  }),
};

File: src/snippets/rate-limiting/reference/python-flask/ByUserId.py
import os

from arcjet import Mode, arcjet_sync, fixed_window
from flask import Flask, jsonify, request

app = Flask(__name__)

aj = arcjet_sync(
    key=os.environ["ARCJET_KEY"],  # Get your site key from https://app.arcjet.com
    rules=[
        fixed_window(
            mode=Mode.LIVE,
            characteristics=["userId"],
            window=3600,  # 1 hour in seconds
            max=60,
        ),
    ],
)


@app.get("/")
def index():
    # Pass userId as a string to identify the user. This could also be a number
    # or boolean value
    decision = aj.protect(request, characteristics={"userId": "user123"})
    print("Arcjet decision", decision)

    if decision.is_denied():
        return jsonify(error="Too Many Requests"), 429

    return jsonify(message="Hello world")
